timestep.c

```diff
@@ -169,10 +169,15 @@ find_timesteps(void)
     /*Update the PM timestep size */
     if(is_PM_timestep(All.Ti_Current)) {
         SyncPoint * next = find_next_sync_point(All.Ti_Current);
-        if(next == NULL) 
-            endrun(0, "no more steps; shouldn't happen unless the main loop is messed up \n");
-        /* go no more than the next sync point */
-        inttime_t dti_max = next->ti - PM.Ti_kick;
+        inttime_t dti_max;
+        if(next == NULL) {
+            message(0, "Trying to go beyond the last sync point. This happens when TimeMax is less or equal to the last OutputTime \n");
```

Comment:
Before this merge, if TimeMax <= last output time we just discard later output times, and make TimeMax the end of the timeline. There was logic in setup_output_times to do that

Reply:
There is still the logic in setup_sync_points to do that.

But it is still possible to run out of sync point when we are at TimeMax -- this is actually how we can logically infer the run shall terminate.

I was expecting to see a warning like this at and only at TimeMax. and on travis it did this. So it is OK to remove it.

Reply:
got it.

Reply:
I "cleaned" this up in the next commit.